compiler/rustc_codegen_llvm/src/declare.rs

```diff
@@ -68,6 +68,15 @@ pub(crate) fn declare_raw_fn<'ll, 'tcx>(
     debug!("declare_raw_fn(name={:?}, ty={:?})", name, ty);
     let llfn = declare_simple_fn(cx, name, callconv, unnamed, visibility, ty);
 
+    if cx.sess().target.has_convergent {
+        // Conservatively apply convergent to all functions
+        attributes::apply_to_llfn(
+            llfn,
+            llvm::AttributePlace::Function,
+            &[llvm::AttributeKind::Convergent.create_attr(cx.llcx)],
+        );
+    }
```

Comment:
Maybe put this in `attributes::apply_to_llfn` instead?

Reply:
Could you clarify what you mean?
I don’t see much to change here
1. I can’t put adding the convergent attribute anywhere lower-level as it requires knowing the target we are compiling for (we don’t want to pessimize non-gpu targets)
2. `apply_to_llfn` is used for function, parameter and return attributes, so the AttributePlace cannot be removed
3. Removing the `create_attr` isn’t trivial, `apply_to_llfn` would need to create a converted vector and there are some places where we directly put in an `Attribute` instead of an `AttributeKind`

Reply:
Yes, good point, I’ll do that.